diagrams/HighLevelStateDiagram.md

```diff
@@ -0,0 +1,131 @@
+
+//All states
+```plantuml
+@startuml
+hide empty description
+
+[*] --> FollowerState
```

Comment:
Raft allows client to be at most in single state [1]:
- Follower
- Candidate
- Leader

Each state transition must have a motivation, essentially answering a question, why system moved from one state to another? Eg https://gist.github.com/jnick26/54bf6dda9dabd7e075495943210fc85d

[1] 5.1 Raft basics, https://raft.github.io/raft.pdf